runtime/src/accounts_db.rs

```diff
@@ -2345,8 +2352,13 @@ impl AccountsDb {
         max_root: Option<Slot>,
         clone_in_lock: bool,
     ) -> Option<(Slot, AppendVecId, usize, Option<LoadedAccountAccessor<'a>>)> {
-        let (lock, index) = self.accounts_index.get(pubkey, Some(ancestors), max_root)?;
-        // Notice the subtle `?` at previous line, we bail out pretty early for missing.
+        let (lock, index) = match self.accounts_index.get(pubkey, Some(ancestors), max_root) {
+            AccountIndexGetResult::Success(lock, index) => (lock, index),
+            // we bail out pretty early for missing.
+            _ => {
```

Comment:
Might be worth explicitly listing out the options here, in case we add something else in the future

Reply:
I added the other 2 options.